python/lsst/validate/drp/checkAstrometryPhotometry.py

```diff
@@ -0,0 +1,469 @@
+    """
+    ra_avg, dec_avg = averageRaDec(cat)
+    ra, dec = cat.get('coord_ra'), cat.get('coord_dec')
+    # Approximating that the cos(dec) term is roughly the same
+    #   for all observations of this object.
+    ra_diff  = (ra - ra_avg) * np.cos(dec)**2
+    dec_diff = dec - dec_avg
+    pos_diff = np.sqrt(ra_diff**2 + dec_diff**2)  # radians
+    pos_diff = [afwGeom.radToMas(p) for p in pos_diff]  # milliarcsec
+
+    return pos_diff
+
+def positionRms(cat):
+    """Calculate the RMS for RA, Dec for a set of observations an object.
+
+    @param[in]  cat -- Collection with a .get method 
+         for 'coord_ra', 'coord_dec' that returns radians.
+
+    @param[out]  pos_rms -- RMS of positions in milliarcsecond.  Float.
+
+    This is WRONG!
+    Doesn't do wrap-around
+    """
+    ra_avg, dec_avg = averageRaDec(cat)
+    ra, dec = cat.get('coord_ra'), cat.get('coord_dec')
+    # Approximating that the cos(dec) term is roughly the same
+    #   for all observations of this object.
+    ra_var = np.var(ra) * np.cos(dec_avg)**2
+    dec_var = np.var(dec)
+    pos_rms = np.sqrt(ra_var + dec_var)  # radians
+    pos_rms = afwGeom.radToMas(pos_rms)  # milliarcsec
+
+    return pos_rms
+
+
+def loadAndMatchData(repo, visitDataIds,
+                     matchRadius=afwGeom.Angle(1, afwGeom.arcseconds)):
+    """Load data from specific visit.  Match with reference.
+
+    @param repo  The repository.  This is generally the directory on disk
+                    that contains the repository and mapper.
+    @param visitDataIds   Butler Data ID of Image catalogs to compare to reference.
+           The actual pixel image is also needed for now for the photometric calibration.
+           List.
+
+    @param matchRadius    Radius for matching.  afwGeom.Angle().
+
+    Return a afw.table.GroupView object of matched catalog.
+    """
+
+    # Following 
+    # https://github.com/lsst/afw/blob/tickets/DM-3896/examples/repeatability.ipynb
+    butler = dafPersist.Butler(repo)
+    dataset = 'src'
+
+    dataRefs = [dafPersist.ButlerDataRef(butler, vId) for vId in visitDataIds]
+
+    # retrieve the schema of the source catalog and extend it in order to add a field to record the ccd number
+    ccdKeyName = getCcdKeyName(visitDataIds[0])
+
+    schema = butler.get(dataset + "_schema", immediate=True).schema
+    mapper = SchemaMapper(schema)
+    mapper.addMinimalSchema(schema)
+#    mapper.addOutputField(Field[int](ccdKeyName, "CCD number"))
+    mapper.addOutputField(Field[float]('base_PsfFlux_mag', "PSF magnitude"))
+    mapper.addOutputField(Field[float]('base_PsfFlux_magerr', "PSF magnitude uncertainty"))
+    newSchema = mapper.getOutputSchema()
+
+    # Create an object that can match multiple catalogs with the same schema
+    mmatch = MultiMatch(newSchema, 
+                        dataIdFormat = {'visit': int, ccdKeyName: int},
+                        radius=matchRadius,
+                        RecordClass=SimpleRecord)
+
+    if False:
```

Comment:
Again, please don't leave disabled code in the committed source unless there's a good reason (and that reason is well explained).